### DISCUSSION TITLE
Is it possible to adjust fans "curve" for performance/auto/quite modes?

### DISCUSSION THREAD

**Discussion by pripishchik** [BartoszCichecki/LenovoLegionToolkit#710] [Q&A] [answered]
Currently I see that is the only way to change fans speed is to select "Custom" power mode, but I don't know how to setup "Custom" power mode to match "Performance" or "Balance" modes. I would like to just tweak fans (fans "curve") without changing the power mode. Or how to setup "Custom" power mode to match "Performance mode" in terms of "wats" etc., so I can then just mimic "Performance" mode with "Custom" but with the ability to tweak fans?

**Comment by BartoszCichecki** [comment, marked as answer]
Lenovo did not give an option to control fans in other modes.

**Reply by pripishchik** [reply]
thanks, and is there any way to "copy" "Performance" mode settings into "Custom" mode (to mimic "Performance" under "Custom")?

**Reply by BartoszCichecki** [reply]
If your machine supports it, there will be a button in bottom left corner of the custom mode settings windows. If there is no button, you can check power limits in HWInfo and just apply them manually.

**Reply by pripishchik** [reply]
yes, I see the button, cool. but does this button take "Performance" mode settings from my laptop "Performance" mode or this is some general "Performance" mode settings for all laptops?

**Reply by BartoszCichecki** [reply]
Performance mode, but again you can always verify with HWINFO; mistakes in reported values happen.

**Reply by pripishchik** [reply]
yes, but there is no **CPU Long Term Power Limit (Cross loading)**, **PL1 Tau**, **GPU cTGP**, **Total Processor Power Target In AC** in HWINFO.

**Reply by BartoszCichecki** [reply]
Not sure, what answer do you expect. Values _should_ be from the actual Performance mode. They are read from the API that is exposed by Lenovo.

**Reply by pripishchik** [reply]
yes, and Lenovo Legion Toolkit reads this API I guess, but for some reason there is some inconsistency between values from the API and from HWINFO.

e.g. **CPU Platform Short Term Power Limit** is 185W in HWINFO when actual "Performance" mode is selected, but 175W when "Custom" mode selected with loaded "Performance" mode from the corresponding dropdown button. Why? Lenovo API is not reliable?

**Reply by BartoszCichecki** [reply]
Not sure I understand, but it might be that Lenovo is using 185W for PL1 in Performance, but the function that returns power limits in different power modes, returns 175W. That wouldn't be the first time there differences like this on their side.
